ark/utils/spatial_analysis_utils.py

```diff
@@ -207,15 +221,24 @@ def compute_close_cell_num_random(marker_nums, dist_mat, dist_lim, bootstrap_num
     close_num_rand = np.zeros((
         len(marker_nums), len(marker_nums), bootstrap_num), dtype='int')
 
-    dist_bin = np.zeros(dist_mat.shape)
-    dist_bin[dist_mat < dist_lim] = 1
+    # dist_bin = np.zeros(dist_mat.shape, dtype='int')
+    # dist_bin[dist_mat.values < dist_lim] = 1
+
+    # dist_bin = xr.DataArray(dist_bin,
+    #                         coords=[dist_mat.coords, dist_mat.coords])
+
+    dist_mat_bin = xr.DataArray(
+        (dist_mat.values < dist_lim).astype(np.int8),
+        coords=dist_mat.coords
+    )
 
     for j, m1n in enumerate(marker_nums):
         for k, m2n in enumerate(marker_nums[j:], j):
             close_num_rand[j, k, :] = np.sum(
-                np.random.choice(dist_bin.flatten(), (m1n * m2n, bootstrap_num), True),
+                np.random.choice(dist_mat_bin.values.flatten(), (m1n * m2n, bootstrap_num), True),
```

Comment:
Sam thing here, one liners are great but not if they make the code too complex. Specifying the output shape on one line, then using that output shape would be better than defining the output shape in the function (I think)